modules/issue_tracker/php/edit.class.inc

```diff
@@ -0,0 +1,876 @@
+     *
+     * @return array $commentHistory
+     */
+    function getComments(int $issueID): array
+    {
+        $factory = \NDB_Factory::singleton();
+        $db      = $factory->database();
+
+        $unformattedComments = $db->pselect(
+            "SELECT newValue, fieldChanged, dateAdded, addedBy " .
+            "FROM issues_history where issueID=:issueID " .
+            "UNION " .
+            "SELECT issueComment, 'comment', dateAdded, addedBy " .
+            "FROM issues_comments where issueID=:issueID ",
+            ['issueID' => $issueID]
+        );
+
+        // looping by reference so can edit in place
+        $modules = \Module::getActiveModulesIndexed($db);
+        foreach ($unformattedComments as &$comment) {
+            if ($comment['fieldChanged'] === 'module') {
+                $mid = $comment['newValue'];
+                $comment['newValue'] = $modules[$mid]->getLongName();
+            } else if ($comment['fieldChanged'] === 'centerID') {
+                $comment['newValue']     = $this->getSiteName(
+                    intval($comment['newValue'])
+                );
+                $comment['fieldChanged'] = 'site';
+            } else if ($comment['fieldChanged'] === 'candID') {
+                $PSCID = $db->pselectOne(
+                    "SELECT PSCID FROM candidate WHERE CandID=:candID",
+                    ['candID' => $comment['newValue']]
+                );
+                $comment['newValue']     = $PSCID;
+                $comment['fieldChanged'] = 'PSCID';
+            } else if ($comment['fieldChanged'] === 'sessionID') {
+                $visitLabel          = $db->pselectOne(
+                    "SELECT Visit_label FROM session WHERE ID=:sessionID",
+                    ['sessionID' => $comment['newValue']]
+                );
+                $comment['newValue'] = $visitLabel;
+                $comment['fieldChanged'] = 'Visit Label';
+            }
+        }
+
+        return $unformattedComments; //now formatted I guess
+    }
+
+    /**
+     * Gets a list of those watching an issue.
+     *
+     * @param int $issueID the relevant issue
+     *
+     * @return array those who are watching
+     */
+    function getWatching(int $issueID): array
+    {
+        $factory = \NDB_Factory::singleton();
+        $db      = $factory->database();
+
+        $watching = $db->pselect(
+            "SELECT userID from issues_watching WHERE issueID=:issueID",
+            ['issueID' => $issueID]
+        );
+
+        $whoIsWatching = [];
+        foreach ($watching as $watcher) {
+            $whoIsWatching[] = $watcher['userID'];
+        }
+        return $whoIsWatching;
+    }
+
+    /**
+     * Returns the site name in the database corresponding to the centerID
+     *
+     * @param int $centerID - must match one from the psc table or 0
+     *
+     * @return ?string
+     */
+    function getSiteName(int $centerID): ?string
+    {
+        $db =& \Database::singleton();
```

Comment:
I don't see why this should use the database instead of a Site object, since you have the $centerID to instantiate it.